component/component.go

```diff
@@ -40,6 +51,18 @@ type Host interface {
 	// Context returns a context provided by the host to be used on the component
 	// operations.
 	Context() context.Context
```

Comment:
Unrelated to this PR, but I am not sure I understand this Context function.

Reply:
@pjanotti may be able to explain, I don't remember.

Reply:
@pjanotti can you take this as an action item to update the comment and explain what is that function for and how is it used.

Reply:
@bogdandrutu I'm unsure but this looks like a leftover from when some components needed a common context to start their operations and this was passed via the `Host`. It will make more sense to remove it from here and add it to Start. I will check the appropriate alternative.